commit ef76edc2e27e49e7e148894948c8dda394b796c1
Author: vmakarov <vmakarov@138bc75d-0d04-0410-961f-82ee72b054a4>
Date:   Fri Jul 11 17:19:53 2014 +0000

    2014-07-11  Vladimir Makarov  <vmakarov@redhat.com>
    
    	* lra-constraints.c (remove_inheritance_pseudos): Process
    	destination pseudo too.
    
    
    
    git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@212464 138bc75d-0d04-0410-961f-82ee72b054a4

 	* gcov-tool.c (gcov_output_files): Fix build error introduced in
diff --git a/gcc/lra-constraints.c b/gcc/lra-constraints.c
index baed7e7..a43f8dc 100644
--- a/gcc/lra-constraints.c
+++ b/gcc/lra-constraints.c
@@ -5697,6 +5697,20 @@ remove_inheritance_pseudos (bitmap remove_pseudos)
 			SUBREG_REG (SET_SRC (set)) = SET_SRC (prev_set);
 		      else
 			SET_SRC (set) = SET_SRC (prev_set);
+		      /* As we are finishing with processing the insn
+			 here, check the destination too as it might
+			 inheritance pseudo for another pseudo.  */
+		      if (bitmap_bit_p (remove_pseudos, dregno)
+			  && bitmap_bit_p (&lra_inheritance_pseudos, dregno)
+			  && (restore_regno
+			      = lra_reg_info[dregno].restore_regno) >= 0)
+			{
+			  if (GET_CODE (SET_DEST (set)) == SUBREG)
+			    SUBREG_REG (SET_DEST (set))
+			      = regno_reg_rtx[restore_regno];
+			  else
+			    SET_DEST (set) = regno_reg_rtx[restore_regno];
+			}
 		      lra_push_insn_and_update_insn_regno_info (curr_insn);
 		      lra_set_used_insn_alternative_by_uid
 			(INSN_UID (curr_insn), -1);
